src/hammer-vlsi/hammer_vlsi/hammer_vlsi_impl.py

```diff
@@ -1476,10 +1476,10 @@ def sdc_clock_constraints(self) -> str:
                     groups[clock.group] = [clock.name]
             else:
                 ungrouped_clocks.append(clock.name)
-        if len(groups):
+        if len(groups) + len(ungrouped_clocks):
```

Comment:
Isn't this just equivalent to `if len(clocks):` now?
Basically we want to do this as long as there are any clocks in the system.
The single clock or single group design will be treated as a no-op but I think for simplicity of this code that is fine.

Reply:
Yeah `len(clocks)` is neater.